static/app/views/dashboards/widgets/timeSeriesWidget/releaseBubbles/utils/createReleaseBuckets.tsx

```diff
@@ -0,0 +1,74 @@
+import type {ReleaseMetaBasic} from 'sentry/types/release';
+import type {Bucket} from 'sentry/views/dashboards/widgets/timeSeriesWidget/releaseBubbles/types';
+
+//
+// Note: You probably don't want to use this directly, use `useReleaseBubbles`
+// instead.
+//
+// Create `desiredBuckets` number of release buckets using the
+// min()/max() of timestamps in all `timeSeries`'s `data`.
+//
+// The timestamp inclusivity is as follows:
+//
+// [
+//  [inclusive, inclusive],
+//  [exclusive, inclusive],
+//  ...
+//  [exclusive, inclusive],
+// ]
+//
+// where only the first bucket's starting timestamp is inclusive.
+//
+export function createReleaseBuckets(
+  minTime: number | undefined,
+  maxTime: number | undefined,
+  releases: ReleaseMetaBasic[],
+  desiredBuckets = 10
+): Bucket[] {
+  const buckets: Bucket[] = [];
+
+  if (!minTime || !maxTime) {
+    return [];
+  }
+
+  // We need to create <desiredBuckets> number of buckets between
+  // [minDate, maxDate]. Last bucket always ends at maxDate and does not
+  // necessarily have the same bucket width as the other buckets.
+  const timeDiff = maxTime - minTime;
+  const interval = Math.ceil(timeDiff / desiredBuckets);
+
+  for (let i = 0; i < desiredBuckets; i++) {
+    const bucketStartTs = minTime + i * interval;
+    // Ending timestamp will clump in the remaining bits if it's not
+    // evenly distributed
+    const bucketEndTs = i === desiredBuckets - 1 ? maxTime : bucketStartTs + interval;
+    buckets.push([bucketStartTs, 0, bucketEndTs, 0, []]);
```

Comment:
I think if `Bucket` was an object, this would be a lot easier to read! I couldn't figure out what `placeholder` was for! Also, below, when setting `currentBucket[2]` it's not clear what's happening, IMO

Reply:
I think this is due to echarts shenanigans unfortunately, but I'll take another peek

Reply:
I was really thrown off by `api.value` api in `renderItem`, turns out we have a closure with the data already and access to the data index (echarts API does not include data afaict)

[`f940ba0` (#85270)](https://github.com/getsentry/sentry/pull/85270/commits/f940ba0e149cdb446df2fe9cebf06e8ef9dd1c68)